pkg/reconciler/artifactbuild/artifactbuild.go

```diff
@@ -257,10 +258,38 @@ func (r *ReconcileArtifactBuild) handlePipelineRunReceived(ctx context.Context,
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	result, err2 := RemovePipelineFinalizer(ctx, pr, r.client)
```

Comment:
also, in the dependencybuild.go versions of this, if the ownerfefs were missing, you removed the finalizer:
```
	ownerRefs := pr.GetOwnerReferences()
	if len(ownerRefs) == 0 {
		msg := "pipelinerun missing ownerrefs %s:%s"
		r.eventRecorder.Eventf(pr, v1.EventTypeWarning, msg, pr.Namespace, pr.Name)
		log.Info(msg, pr.Namespace, pr.Name)
		return artifactbuild.RemovePipelineFinalizer(ctx, pr, r.client)
	}
	ownerName := ""
	for _, ownerRef := range ownerRefs {
		if strings.EqualFold(ownerRef.Kind, "dependencybuild") || strings.EqualFold(ownerRef.Kind, "dependencybuilds") {
			ownerName = ownerRef.Name
			break
		}
	}
	if len(ownerName) == 0 {
		msg := "pipelinerun missing dependencybuild ownerrefs %s:%s"
		r.eventRecorder.Eventf(pr, v1.EventTypeWarning, "MissingOwner", msg, pr.Namespace, pr.Name)
		log.Info(msg, pr.Namespace, pr.Name)

		return artifactbuild.RemovePipelineFinalizer(ctx, pr, r.client)
	}
```
but you are not at the top of this method .... maybe that is because in the ab case, you are certain the finalizer would not have been added?

even if that is the case, adding the call there as well feels like a benign, better safe than sorry sort of thing